### ISSUE TITLE
Repeated replies in chat mode with flexgen

### ISSUE THREAD

**Issue by MetaIX** [oobabooga/text-generation-webui#104]
I've successfully loaded opt-30b-iml-max via python server.py --model opt-iml-max-30b --flexgen --compress-weight --cai-chat --percent 100 0 100 0 100 0 on a 4090. 

Unfortunately, the bot appears to be repeating the same greeting message after I'm 6 messages in. 

![Untitled](https://user-images.githubusercontent.com/125941078/220566578-1c04092f-28a8-4819-9cd1-43c463294b0b.png)

Also, trying to switch bots when this happens gives me a seemingly infinite loading time.

![image](https://user-images.githubusercontent.com/125941078/220567397-af7ae0b4-fcc3-4bbe-80e6-63c6cbba43d5.png)

No errors appear on the console.

![image](https://user-images.githubusercontent.com/125941078/220567614-ab397168-122e-4abf-892f-c6e99eba53b5.png)

![image](https://user-images.githubusercontent.com/125941078/220568376-8c8b1f40-aad5-4080-9afb-a4f2cde9c4c0.png)

What could be causing this? Happens with multiple bots.

**Comment by MetaIX** [comment]
I'm just testing behaviors with different models. Honestly, for the 2 messages it managed to answer, it wasn't bad at all. 0.09 there must be something wrong. I'll also test opt-30b soon hopefully I'll get similar speeds to what I have.

**Comment by MetaIX** [comment]
That's interesting. I wonder what could be causing it then.

**Comment by oobabooga** [comment]
This looks like a silent CUDA out of memory error. I will make some experiments with 30b models later and will report my findings.

**Comment by Ph0rk0z** [comment]
For me it helps to put generation functions on CPU like --percent xx xx 0 100 0 100. Then memory doesn't increase while generating anymore.. I can shave 10s off of normal offloading on the 13b model. I have not tried the 30b yet but I doubt results will be different. Just my % gpu will be lower.


From opt13b
Output generated in 27.44 seconds (0.36 it/s, 80 tokens)
Output generated in 26.59 seconds (0.38 it/s, 80 tokens)
Output generated in 30.43 seconds (0.33 it/s, 80 tokens)

It also generates the token limit 100% of the time. This is definitely slowing it down.

**Comment by oobabooga** [comment]
@Ph0rk0z it is true that the token limit is being generated 100% of the time in flexgen. This is really annoying. I seem to be using the `stop` parameter exactly as in the official FlexGen chatbot example, but it doesn't do anything.

**Comment by oobabooga** [comment]
I think that this fixes the issue of FlexGen not stopping the generation at a new line character

https://github.com/oobabooga/text-generation-webui/commit/6e843a11d64ec0898a1cb6f2cc9a81619038db81

**Comment by Ph0rk0z** [comment]
Yes. I agree. This is fixed now after API changes and updates above.

**Comment by mcastillobalderas** [comment]
> Cargué con éxito opt-30b-iml-max a través de python server.py --model opt-iml-max-30b --flexgen --compress-weight --cai-chat --percent 100 0 100 0 100 0 en un 4090.
> 
> Desafortunadamente, el bot parece estar repitiendo el mismo mensaje de saludo después de 6 mensajes.
> 
> ![Intitulado](https://user-images.githubusercontent.com/125941078/220566578-1c04092f-28a8-4819-9cd1-43c463294b0b.png)
> 
> Además, tratar de cambiar de bots cuando esto sucede me da un tiempo de carga aparentemente infinito.
> 
> ![imagen](https://user-images.githubusercontent.com/125941078/220567397-af7ae0b4-fcc3-4bbe-80e6-63c6cbba43d5.png)
> 
> No aparecen errores en la consola.
> 
> ![imagen](https://user-images.githubusercontent.com/125941078/220567614-ab397168-122e-4abf-892f-c6e99eba53b5.png)
> 
> ![imagen](https://user-images.githubusercontent.com/125941078/220568376-8c8b1f40-aad5-4080-9afb-a4f2cde9c4c0.png)
> 
> ¿Qué podría estar causando esto? Ocurre con múltiples bots.

Abierto
